### ISSUE TITLE
利用sd3.5训练图像生成，经过36epoch依然很模糊

### ISSUE THREAD

**Issue by WangzcBruce** [SenseTime-FVG/OpenDWM#64]
![Image](https://github.com/user-attachments/assets/1f96618c-a1da-49e5-9e78-bd38e1c65837)
{
    "device": "cuda",
    "ddp_backend": "nccl",
    "train_epochs": 36,
    "generator_seed": 0,
    "data_shuffle": true,
    "global_state": {
        "nuscenes_fs": {
            "_class_name": "dwm.fs.czip.CombinedZipFileSystem",
            "fs": {
                "_class_name": "dwm.fs.dirfs.DirFileSystem",
                "path": "/workspace/group_share/adc-perception-xbrain/public_dataset/nuscenes/converted"
            },
            "paths": [
                "interp_12Hz_trainval.zip",
                "v1.0-trainval_meta.zip",
                "v1.0-trainval01_blobs.zip",
                "v1.0-trainval02_blobs.zip",
                "v1.0-trainval03_blobs.zip",
                "v1.0-trainval04_blobs.zip",
                "v1.0-trainval05_blobs.zip",
                "v1.0-trainval06_blobs.zip",
                "v1.0-trainval07_blobs.zip",
                "v1.0-trainval08_blobs.zip",
                "v1.0-trainval09_blobs.zip",
                "v1.0-trainval10_blobs.zip",
                "nuScenes-map-expansion-v1.3.zip"
            ]
        }
    },
    "optimizer": {
        "_class_name": "torch.optim.AdamW",
        "lr": 6e-5
    },
    "pipeline": {
        "_class_name": "dwm.pipelines.ctsd.CrossviewTemporalSD",
        "common_config": {
            "disable_crossview": false,
            "disable_temporal": true,
            "cat_condition": true,
            "cond_with_action": false,
            "condition_on_all_frames": true,
            "added_time_ids": "fps_camera_transforms",
            "camera_intrinsic_embedding_indices": [
                0,
                4,
                2,
                5
            ],
            "camera_intrinsic_denom_embedding_indices": [
                1,
                1,
                0,
                1
            ],
            "camera_transform_embedding_indices": [
                2,
                6,
                10,
                3,
                7,
                11
            ],
            "distribution_framework": "fsdp",
            "ddp_wrapper_settings": {
                "sharding_strategy": {
                    "_class_name": "torch.distributed.fsdp.ShardingStrategy",
                    "value": 5
                },
                "auto_wrap_policy": {
                    "_class_name": "torch.distributed.fsdp.wrap.ModuleWrapPolicy",
                    "module_classes": [
                        {
                            "_class_name": "get_class",
                            "class_name": "diffusers.models.attention.JointTransformerBlock"
                        },
                        {
                            "_class_name": "get_class",
                            "class_name": "dwm.models.crossview_temporal.VTSelfAttentionBlock"
                        }
                    ]
                },
                "mixed_precision": {
                    "_class_name": "torch.distributed.fsdp.MixedPrecision",
                    "param_dtype": {
                        "_class_name": "get_class",
                        "class_name": "torch.float16"
                    }
                }
            },
            "t5_fsdp_wrapper_settings": {
                "sharding_strategy": {
                    "_class_name": "torch.distributed.fsdp.ShardingStrategy",
                    "value": 4
                },
                "auto_wrap_policy": {
                    "_class_name": "torch.distributed.fsdp.wrap.ModuleWrapPolicy",
                    "module_classes": [
                        {
                            "_class_name": "get_class",
                            "class_name": "transformers.models.t5.modeling_t5.T5Block"
                        }
                    ]
                }
            },
            "text_encoder_load_args": {
                "variant": "fp16",
                "torch_dtype": {
                    "_class_name": "get_class",
                    "class_name": "torch.float16"
                }
            }
        },
        "training_config": {
            "freezing_pattern": "^(transformer_blocks|time_text_embed)$",
            "text_prompt_condition_ratio": 0.8,
            "3dbox_condition_ratio": 0.8,
            "hdmap_condition_ratio": 0.8,
            "enable_grad_scaler": true
        },
        "inference_config": {
            "guidance_scale": 4,
            "inference_steps": 40,
            "preview_image_size": [
                448,
                252
            ],
            "evaluation_item_count": 1664
        },
        "model": {
            "_class_name": "dwm.models.crossview_temporal_dit.DiTCrossviewTemporalConditionModel",
            "dual_attention_layers": [
                0,
                1,
                2,
                3,
                4,
                5,
                6,
                7,
                8,
                9,
                10,
                11,
                12
            ],
            "attention_head_dim": 64,
            "caption_projection_dim": 1536,
            "in_channels": 16,
            "joint_attention_dim": 4096,
            "num_attention_heads": 24,
            "num_layers": 24,
            "out_channels": 16,
            "patch_size": 2,
            "pooled_projection_dim": 2048,
            "pos_embed_max_size": 384,
            "qk_norm": "rms_norm",
            "qk_norm_on_additional_modules": "rms_norm",
            "sample_size": 128,
            "perspective_modeling_type": "implicit",
            "projection_class_embeddings_input_dim": 2816,
            "enable_crossview": true,
            "crossview_attention_type": "rowwise",
            "crossview_block_layers": [
                1,
                5,
                9,
                13,
                17,
                21
            ],
            "crossview_gradient_checkpointing": true,
            "enable_temporal": false,
            "temporal_gradient_checkpointing": true,
            "mixer_type": "AlphaBlender",
            "merge_factor": 2,
            "condition_image_adapter_config": {
                "in_channels": 6,
                "channels": [
                    1536,
                    1536,
                    1536,
                    1536,
                    1536,
                    1536
                ],
                "is_downblocks": [
                    true,
                    false,
                    false,
                    false,
                    false,
                    false
                ],
                "num_res_blocks": 2,
                "downscale_factor": 8,
                "use_zero_convs": true
            }
        },
        "pretrained_model_name_or_path": "/workspace/wangzc13@xiaopeng.com/stable-diffusion-3.5-medium",
        "model_checkpoint_path": "/workspace/wangzc13@xiaopeng.com/stable-diffusion-3.5-medium/sd3.5_medium.safetensors",
        "model_load_state_args": {
            "strict": false
        },
        "metrics": {
            "fid": {
                "_class_name": "torchmetrics.image.fid.FrechetInceptionDistance",
                "normalize": true
            }
        }
    },
    "training_dataset": {
        "_class_name": "dwm.datasets.common.DatasetAdapter",
        "base_dataset": {
            "_class_name": "dwm.datasets.nuscenes.MotionDataset",
            "fs": {
                "_class_name": "dwm.common.get_state",
                "key": "nuscenes_fs"
            },
            "dataset_name": "interp_12Hz_trainval",
            "split": "train",
            "sequence_length": 1,
            "fps_stride_tuples": [
                [
                    0,
                    1
                ]
            ],
            "sensor_channels": [
                "CAM_FRONT_LEFT",
                "CAM_FRONT",
                "CAM_FRONT_RIGHT",
                "CAM_BACK_RIGHT",
                "CAM_BACK",
                "CAM_BACK_LEFT"
            ],
            "keyframe_only": true,
            "enable_camera_transforms": true,
            "_3dbox_image_settings": {
                "color_table": {
                    "vehicle.car": [
                        0,
                        0,
                        255
                    ],
                    "vehicle.bus": [
                        128,
                        0,
                        255
                    ],
                    "vehicle.truck": [
                        255,
                        255,
                        0
                    ],
                    "vehicle.construction": [
                        128,
                        128,
                        255
                    ],
                    "vehicle.emergency": [
                        255,
                        128,
                        128
                    ],
                    "vehicle.trailer": [
                        255,
                        255,
                        255
                    ],
                    "vehicle.bicycle": [
                        128,
                        255,
                        0
                    ],
                    "vehicle.motorcycle": [
                        0,
                        255,
                        128
                    ],
                    "human.pedestrian": [
                        255,
                        0,
                        0
                    ],
                    "movable_object.barrier": [
                        64,
                        64,
                        64
                    ],
                    "movable_object.trafficcone": [
                        128,
                        128,
                        0
                    ]
                }
            },
            "hdmap_image_settings": {
                "hdmap_images_perldiff": false,
                "hdmap_images_official": false,
                "hdmap_images_perldiff_adjust": false,
                "hdmap_images_perldiff_8kinds": false,
                "hdmap_images_perldiff_3kinds": true
            },
            "image_description_settings": {
                "path": "/workspace/group_share/adc-perception-xbrain/public_dataset/nuscenes/nuscenes_v1.0-trainval_caption_v2_train.json",
                "time_list_dict_path": "/workspace/group_share/adc-perception-xbrain/public_dataset/nuscenes/nuscenes_v1.0-trainval_caption_v2_times_train.json",
                "align_keys": [
                    "time",
                    "weather"
                ],
                "reorder_keys": true,
                "drop_rates": {
                    "environment": 0.04,
                    "objects": 0.08,
                    "image_description": 0.16
                }
            },
            "stub_key_data_dict": {
                "crossview_mask": [
                    "content",
                    {
                        "_class_name": "torch.tensor",
                        "data": {
                            "_class_name": "json.loads",
                            "s": "[[1,1,0,0,0,1],[1,1,1,0,0,0],[0,1,1,1,0,0],[0,0,1,1,1,0],[0,0,0,1,1,1],[1,0,0,0,1,1]]"
                        },
                        "dtype": {
                            "_class_name": "get_class",
                            "class_name": "torch.bool"
                        }
                    }
                ]
            }
        },
        "transform_list": [
            {
                "old_key": "images",
                "new_key": "vae_images",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ColorJitter",
                            "brightness": 0.02,
                            "contrast": 0.03,
                            "saturation": 0.03
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
                        {
                "old_key": "3dbox_masks",
                "new_key": "3dbox_masks",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "3dbox_images",
                "new_key": "3dbox_images",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "hdmap_images_perldiff_3kinds",
                "new_key": "hdmap_images_perldiff_3kinds",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "image_description",
                "new_key": "clip_text",
                "transform": {
                    "_class_name": "dwm.datasets.common.Copy"
                },
                "stack": false
            }
        ],
        "pop_list": [
            "images",
            "image_description"
        ]
    },
    "validation_dataset": {
        "_class_name": "dwm.datasets.common.DatasetAdapter",
        "base_dataset": {
            "_class_name": "dwm.datasets.nuscenes.MotionDataset",
            "fs": {
                "_class_name": "dwm.common.get_state",
                "key": "nuscenes_fs"
            },
            "dataset_name": "v1.0-trainval",
            "split": "val",
            "sequence_length": 1,
            "fps_stride_tuples": [
                [
                    0,
                    3
                ]
            ],
            "sensor_channels": [
                "CAM_FRONT_LEFT",
                "CAM_FRONT",
                "CAM_FRONT_RIGHT",
                "CAM_BACK_RIGHT",
                "CAM_BACK",
                "CAM_BACK_LEFT"
            ],
            "keyframe_only": true,
            "enable_camera_transforms": true,
            "_3dbox_image_settings": {
                "color_table": {
                    "vehicle.car": [
                        0,
                        0,
                        255
                    ],
                    "vehicle.bus": [
                        128,
                        0,
                        255
                    ],
                    "vehicle.truck": [
                        255,
                        255,
                        0
                    ],
                    "vehicle.construction": [
                        128,
                        128,
                        255
                    ],
                    "vehicle.emergency": [
                        255,
                        128,
                        128
                    ],
                    "vehicle.trailer": [
                        255,
                        255,
                        255
                    ],
                    "vehicle.bicycle": [
                        128,
                        255,
                        0
                    ],
                    "vehicle.motorcycle": [
                        0,
                        255,
                        128
                    ],
                    "human.pedestrian": [
                        255,
                        0,
                        0
                    ],
                    "movable_object.barrier": [
                        64,
                        64,
                        64
                    ],
                    "movable_object.trafficcone": [
                        128,
                        128,
                        0
                    ]
                }
            },
            "hdmap_image_settings": {
                "hdmap_images_perldiff": false,
                "hdmap_images_official": false,
                "hdmap_images_perldiff_adjust": false,
                "hdmap_images_perldiff_8kinds": false,
                "hdmap_images_perldiff_3kinds": true
            },
            "image_description_settings": {
                "path": "/workspace/group_share/adc-perception-xbrain/public_dataset/nuscenes/nuscenes_v1.0-trainval_caption_v2_val.json",
                "time_list_dict_path": "/workspace/group_share/adc-perception-xbrain/public_dataset/nuscenes/nuscenes_v1.0-trainval_caption_v2_times_val.json",
                "align_keys": [
                    "time",
                    "weather"
                ]
            },
            "stub_key_data_dict": {
                "crossview_mask": [
                    "content",
                    {
                        "_class_name": "torch.tensor",
                        "data": {
                            "_class_name": "json.loads",
                            "s": "[[1,1,0,0,0,1],[1,1,1,0,0,0],[0,1,1,1,0,0],[0,0,1,1,1,0],[0,0,0,1,1,1],[1,0,0,0,1,1]]"
                        },
                        "dtype": {
                            "_class_name": "get_class",
                            "class_name": "torch.bool"
                        }
                    }
                ]
            }
        },
        "transform_list": [
            {
                "old_key": "images",
                "new_key": "vae_images",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
                        {
                "old_key": "3dbox_masks",
                "new_key": "3dbox_masks",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "3dbox_images",
                "new_key": "3dbox_images",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "hdmap_images_perldiff_3kinds",
                "new_key": "hdmap_images_perldiff_3kinds",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "image_description",
                "new_key": "clip_text",
                "transform": {
                    "_class_name": "dwm.datasets.common.Copy"
                },
                "stack": false
            }
        ],
        "pop_list": [
            "images",
            "image_description"
        ]
    },
    "training_dataloader": {
        "batch_size": 4,
        "num_workers": 3,
        "prefetch_factor": 3,
        "collate_fn": {
            "_class_name": "dwm.datasets.common.CollateFnIgnoring",
            "keys": [
                "clip_text"
            ]
        },
        "persistent_workers": true
    },
    "validation_dataloader": {
        "batch_size": 4,
        "num_workers": 2,
        "prefetch_factor": 3,
        "collate_fn": {
            "_class_name": "dwm.datasets.common.CollateFnIgnoring",
            "keys": [
                "clip_text"
            ]
        },
        "persistent_workers": true
    },
    "preview_dataloader": {
        "batch_size": 2,
        "num_workers": 1,
        "prefetch_factor": 1,
        "shuffle": true,
        "drop_last": true,
        "collate_fn": {
            "_class_name": "dwm.datasets.common.CollateFnIgnoring",
            "keys": [
                "clip_text"
            ]
        },
        "persistent_workers": true
    },
    "informations": {
        "fid": 12.04,
        "total_batch_sizes": 64,
        "steps": 30000
    }
}

**Comment by wzhgba** [comment]
试试增加可训练参数的自由度呢，完全冻结 sd 3.5 原来的权重是不太容易训练的，单视角和多视角的图像分布差异太大了
